File: packages/http-client-java/generator/http-client-generator-clientcore-test/src/test/java/parameters/path/PathTests.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package parameters.path;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

public final class PathTests {

    private final PathClient client = new PathClientBuilder().buildClient();

    @Test
    public void testNormal() {
        client.normal("foo");
    }

    @Disabled("optional path parameter is not supported")
    @Test
    public void testOptional() {
        client.optional();
        client.optional("foo");
    }
}

File: packages/http-client-csharp/generator/TestProjects/Spector.Tests/Http/Parameters/Path/PathTests.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.ClientModel;
using System.ClientModel.Primitives;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Threading;
using System.Threading.Tasks;
using NUnit.Framework;
using Parameters.Path;

namespace TestProjects.Spector.Tests.Http.Parameters.Path
{
    public class PathTests : SpectorTestBase
    {
        [SpectorTest]
        public Task Normal() => Test(async (host) =>
        {
            var client = new PathClient(host, null);
            var name = "foo";
            ClientResult response = await client.NormalAsync(name);
            Assert.AreEqual(204, response.GetRawResponse().Status);
        });

        [SpectorTest]
        [TestCase(true)]
        [TestCase(false)]
        public Task OptionalPathParamIncluded(bool isOptional) => Test(async (host) =>
        {
            var client = new PathClient(host, null);
            string? name = isOptional ? null : "foo";
            ClientResult response = await client.OptionalAsync(name);
            Assert.AreEqual(204, response.GetRawResponse().Status);
        });
    }
}
